test/unittest/models/models_test_utils.cpp

```diff
@@ -113,9 +113,10 @@ class IterationForGolden {
       auto const &lnode = *it;
       auto &rc = lnode->getRunContext();
       for (unsigned int i = 0; i < rc.getNumWeights(); ++i) {
-        if (rc.isWeightDependent(i)) {
+        if (!rc.isGradientFirstAccess(i)) {
```

Comment:
please add a note that we load at the last read of the shared weight, and not at first.

Reply:
I think this is a bug, I'll change like below.

```cpp

    if (rc->isGradientLastAccess(i)) {
      rc->getWeight(i).read(file);
    }
```

Reply:
Done

Reply:
ok.